2022/src/day01.scala

```diff
@@ -0,0 +1,38 @@
+package day01
+
+import scala.util.Using
+import scala.io.Source
+import scala.math.Ordering
+
+import locations.Directory.currentDir
+
+@main def part1: Unit =
+  val input = Using.resource(Source.fromFile(s"$currentDir/../input/day01"))(_.mkString)
+  val answer = part1(input)
+  println(s"The solution is $answer")
+
+@main def part2: Unit =
+  val input = Using.resource(Source.fromFile(s"$currentDir/../input/day01"))(_.mkString)
+  val answer = part2(input)
+  println(s"The solution is $answer")
+
+def part1(input: String): Int =
+  maxInventories(scanInventorys(input.linesIterator), 1).head
+
+def part2(input: String): Int =
+  maxInventories(scanInventorys(input.linesIterator), 3).sum
+
+case class Inventory(items: List[Int])
+
+def scanInventorys(lines: IterableOnce[String]): List[Inventory] =
+  val inventorys = List.newBuilder[Inventory]
+  var items = List.newBuilder[Int]
+  for line <- lines.iterator do
+    if line.isEmpty then
+      inventorys += Inventory(items.result())
+      items = List.newBuilder
+    else items += line.toInt
+  inventorys.result()
```

Comment:
```suggestion
def scanInventories(lines: IterableOnce[String]): List[Inventory] =
  val inventories = List.newBuilder[Inventory]
  var items = List.newBuilder[Int]
  for line <- lines.iterator do
    if line.isEmpty then
      inventories += Inventory(items.result())
      items = List.newBuilder
    else items += line.toInt
  inventories.result()
```

Reply:
Also I find it weird that you call `input.linesInterator` and then `lines.Iterator`.

Could you do something like:
```scala
def scanInventories(input: String): List[Inventory] =
  ...
  for line <- input.linesIterator do
  ...
```

Reply:
good idea